### ISSUE TITLE
Exception on metagenotype_manage page

### ISSUE THREAD

**Issue by kimrutherford** [pombase/canto#1588]
I get this in my browser console.  @jseager7 could you have a look?

```
    at VM3265 canto-modules.js:6246
    at Array.filter (<anonymous>)
    at Scope.$scope.setSingleAllele (VM3265 canto-modules.js:6245)
    at Scope.$scope.setFilters (VM3265 canto-modules.js:6261)
    at Scope.$scope.organismSelected (VM3265 canto-modules.js:6231)
    at fn (eval at compile (VM3250 angular-v1.5.11.js:15126), <anonymous>:4:318)
    at Scope.destination.(anonymous function) [as organismSelected] (http://localhost:7006/static/v1000-v1/js/angular-v1.5.11.js:10279:22)
    at setSelectedOrganism (VM3265 canto-modules.js:3617)
    at VM3265 canto-modules.js:3603
    at VM3250 angular-v1.5.11.js:11728
```

**Comment by jseager7** [comment]
Here's what I get:
```
angular.js:14199 TypeError: Cannot read property 'taxonid' of undefined
    at http://localhost:5000/static/v1247/js/canto-modules.js:6246:76
    at Array.filter (<anonymous>)
    at m.$scope.setSingleAllele (http://localhost:5000/static/v1247/js/canto-modules.js:6245:63)
    at m.$scope.setFilters (http://localhost:5000/static/v1247/js/canto-modules.js:6261:20)
    at m.$scope.organismSelected (http://localhost:5000/static/v1247/js/canto-modules.js:6231:20)
    at fn (eval at compile (http://ajax.googleapis.com/ajax/libs/angularjs/1.5.11/angular.min.js:236:180), <anonymous>:4:318)
    at m.d.(anonymous function) [as organismSelected] (http://ajax.googleapis.com/ajax/libs/angularjs/1.5.11/angular.min.js:84:499)
    at setSelectedOrganism (http://localhost:5000/static/v1247/js/canto-modules.js:3617:12)
    at http://localhost:5000/static/v1247/js/canto-modules.js:3603:7
    at http://ajax.googleapis.com/ajax/libs/angularjs/1.5.11/angular.min.js:98:229
```
Looks like it's happening because of the following function:

https://github.com/pombase/canto/blob/24cc16fd71890faad6f3ee61e61e8d9c1021e012/root/static/js/canto-modules.js#L6244-L6248

When you first navigate to the page, `$scope.data.selectedOrganism` is undefined because no organism is selected in the drop-down lists by default. Also, the error doesn't occur if you only have one host and one pathogen in the session, because the organisms are selected automatically.

Assigning @barnabynorman, since he knows the most about this area of the code.

**Comment by jseager7** [comment]
Ok, it turns out there's a fairly simple fix to this, which is to just add a guard clause to the equality test &ndash; something like:
```javascript
return (
  (e.organism.taxonid === ($scope.data.selectedOrganism.taxonid || -1))
  && (e.alleles.length === 1)
);
```
As far as I know, we can't really ensure that `.taxonid` isn't undefined, because there's no sensible default for the `selectedOrganism` except in the case where there's only one organism: I guess we'll just have to make sure that the consumers of `organismSelector` guard against the fact that `selectedOrganism` can be `undefined`.

**Comment by jseager7** [comment]
The first idea didn't work, but this one does:

```js
$scope.setSingleAllele = function () {
  if ($scope.data.selectedOrganism == null) {
    $scope.data.singleAllele = [];
  } else {
    $scope.data.singleAllele =  $scope.data.genotypes.filter(function (e) {
      return (
        (e.organism.taxonid === $scope.data.selectedOrganism.taxonid)
        && (e.alleles.length === 1)
      );
    });
  }
};
```
Apparently comparing on abstract equality (`==`) can catch both null and undefined, but since this is quite obscure it might be better to be more explicit and use:
```js
if (
    (typeof($scope.data.selectedOrganism) === 'undefined')
    || ($scope.data.selectedOrganism === null)
) {
  $scope.data.singleAllele = [];
}
// etc...
```
Or wrap the check in a function:
```js
var isOrganismSelected = function () {
  return (
    (typeof($scope.data.selectedOrganism) !== 'undefined')
    && ($scope.data.selectedOrganism !== null)
  );
};
```
```js
if (isOrganismSelected()) {
  // filter the organisms
} else {
  $scope.data.singleAllele = [];
}
// etc...
```

**Comment by jseager7** [comment]
This has been fixed by #1590, so I'm closing this now. Feel free to reopen if there any issues with the fix.
